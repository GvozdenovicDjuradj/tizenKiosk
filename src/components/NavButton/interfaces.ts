import { ViewStyle } from "react-native"
import { ButtonProps } from "../Button"

export interface Styles {
  navButton: ViewStyle
  navButtonLeft: ViewStyle
  navButtonRight: ViewStyle
  triangleLeft: ViewStyle
  triangleRight: ViewStyle
  view: ViewStyle
}

export interface NavButtonProps extends ButtonProps {
  buttonColor?: string
  privacyPolicyButtonDefaultText?: string
}

export interface NavButtonNextState {
  isDisabled: boolean
}
