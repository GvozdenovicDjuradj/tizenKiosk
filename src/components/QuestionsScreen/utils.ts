import { Question } from "../../interfaces";

export const getQuestionText = (
  question: Question,
  languageCode?: string
): string =>
  (
    (languageCode && question.questionTranslations
      ? question.questionTranslations.find(
          (item) => item.languageCode === languageCode
        )
      : undefined) || { translation: question.text }
  ).translation;

export const getQuestionAnswerList = (
  question: Question,
  languageCode?: string
): string[] =>
  languageCode &&
  question.multipleChoiceAnswers &&
  question.multipleChoiceAnswers.length > 0
    ? question.multipleChoiceAnswers.map(({ text }) => text)
    : question.selectAnswerList;

export const getQuestionAnswerText = (
  question: Question,
  answerText: string,
  languageCode?: string
): string => {
  const answer =
    languageCode && question.multipleChoiceAnswers
      ? question.multipleChoiceAnswers.find((item) => item.text === answerText)
      : undefined;

  return answer
    ? (
        answer.answerTranslations.find(
          (item) => item.languageCode === languageCode
        ) || { translation: answerText }
      ).translation
    : answerText;
};
