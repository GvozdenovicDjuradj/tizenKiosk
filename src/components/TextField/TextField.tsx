import React from "react"
import { NativeSyntheticEvent, TextInput, TextInputChangeEventData, TextInputProps } from "react-native"
import { input as styles } from "./styles"
import { RootState } from "../../interfaces"
import { connect } from "react-redux"

interface Props extends TextInputProps {
  inputRef?: (input: any) => void;
  fontFamily?: string
  validate?: (e: NativeSyntheticEvent<TextInputChangeEventData>) => void
}

export const TextField = (props: Props) => {
  const { inputRef, fontFamily, style, ...otherProps } = props
  return (
    <TextInput
      underlineColorAndroid="transparent"
      ref={inputRef}
      style={[styles.input, style, { fontFamily } ]}
      onChange={props.validate}
      {...otherProps}
    />
  )
}

const mapStateToProps = (state: RootState) => {
  return {
    fontFamily: state.kiosk.settings && state.kiosk.settings.template.font
  }
}

export default connect(mapStateToProps)(TextField)
