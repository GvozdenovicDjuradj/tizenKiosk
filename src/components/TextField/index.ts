import WrappedInput, { InputProps } from "./WrappedInput"

export { InputProps }

export default WrappedInput
