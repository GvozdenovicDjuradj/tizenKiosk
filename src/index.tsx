import "babel-polyfill"
import { AppRegistry } from "react-native"
import App from "./App"
import app from "../app.json"

AppRegistry.registerComponent(app.name, () => App)

if (window && typeof window === "object" && window.document) {
  AppRegistry.runApplication(app.name, { rootTag: document.getElementById("app") })
}
