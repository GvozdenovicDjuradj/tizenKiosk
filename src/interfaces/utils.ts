import { Action } from "redux"
import { TemplateLanguages } from "./kiosk"

export interface TemplateStyles {
  circle: {
    backgroundColor?: string
  },
  languages: TemplateLanguages,
  logo: string,
  queueIsFullUrl: string,
  showLangSelect: boolean,
  text: {
    color?: string,
    fontFamily?: string,
  }
}

export interface AlertErrorPayload extends Action {
  payload: {
    title: string
    message?: string
  }
}
