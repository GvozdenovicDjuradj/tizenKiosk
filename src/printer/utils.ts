import memoize from "lodash.memoize"
import modelInfo from "./modelInfo.json"
import { CustomerInQueue } from "../interfaces"

export interface ModelInfo {
    names: string[],
    modelTitle: string,
    defaultPaperWidth: number,
    emulation: string,
}

export const searchModelInfoByModelName =
  memoize((modelName: string): ModelInfo | undefined => {

    for (const model of modelInfo.models) {
        const perfectMatch = model.names.find((name: string) => name === modelName)

        if (perfectMatch) {
            return model
        }

        const partialMatch = model.names.find((name: string) => name.startsWith(modelName))

        if (partialMatch) {
            return model
        }

        const includes = model.names.find((name: string) => modelName.toLowerCase().includes(name.toLowerCase()))

        if (includes) {
            return model
        }

        const modelTitleMatch = model.modelTitle === modelName

        if (modelTitleMatch) {
            return model
        }
    }

    return undefined
})

export const testCustomerInQueue: CustomerInQueue = {
    currentPosition: 1,
    customer: {
        bookingRef: "TEST",
        customerProfile: null,
        id: 2,
        identifier: "TEST",
        name: "TEST",
        orderNumber: null,
        ticketNumber: "TEST",
    },
    id: 3,
    minutesRemaining: "TEST",
    queue: {
        averageServeTimeMinutes: 4,
        customerLength: 5,
        id: 6,
        name: "TEST",
        venue: {
            defaultCountryCode: "TEST",
            defaultLanguageIsoCode: "TEST",
            id: 7,
            isBookingEnabled: false,
            isWalkinEnabled: false,
            name: "TEST",
            merchant: {
                featureSettings: {
                    hasQudiniBrand: false,
                },
                id: 8,
            }
        },
    },
    timeRemaining: "TEST",
    joinedTime: "TEST",
    waitTime: 9,
}
