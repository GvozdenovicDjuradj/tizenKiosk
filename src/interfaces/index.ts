export * from "./answers"
export * from "./app"
export * from "./appState"
export * from "./checkIn"
export * from "./customerInQueue"
export * from "./eventCheckIn"
export * from "./kiosk"
export * from "./modal"
export * from "./navigation"
export * from "./product"
export * from "./question"
export * from "./routes"
export * from "./utils"
export * from "./validation"
export * from "./venue"
export * from "./screenSaver"

import { AppState } from "./app"
import { CheckInState } from "./checkIn"
import { EventCheckInState } from "./eventCheckIn"
import { KioskSettingsForm } from "../reducers/kioskSettings"
import { KioskState } from "./kiosk"
import { ModalState } from "./modal"
import { QuestionAnswer } from "./answers"
import { QuestionsReducer } from "./question"
import { ValidationState } from "./validation"
import { PrintingState } from "./printing"
import { PrinterState } from "../printer/reducer"
import { ScreenSaverState } from "./screenSaver"

export interface Flags {
  [key: string]: any
}

export type ValidatorFn = (text: string, ...args: any[]) => { error: string }

export interface RootState {
  answers: QuestionAnswer[];
  app: AppState;
  checkIn: CheckInState;
  error: string;
  eventCheckIn: EventCheckInState;
  kiosk: KioskState;
  kioskSettings: KioskSettingsForm;
  modal: ModalState;
  printer: PrinterState;
  printing: PrintingState;
  questions: QuestionsReducer;
  validation: ValidationState;
  screenSaver?: ScreenSaverState;
}
