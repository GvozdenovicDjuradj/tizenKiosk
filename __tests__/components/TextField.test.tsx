import React from "react"
// import { TextInput, TouchableOpacity, View } from "react-native"
import renderer from "react-test-renderer"
import { Provider } from "react-redux"
import mockStore from "redux-mock-store"
import InputWrapper from "../../src/components/TextField/WrappedInput"
import { TextField } from "../../src/components/TextField/TextField"
// import Popup from "../../src/components/TextField/Popup"
import { AppScreens } from "../../src/interfaces"
import { validators } from "../../src/utils"
// import { VALIDATION_STATE_CHANGE } from "../../src/actions/types"

const createKioskSettingsState = () => ({
  settings: {
    template: {
      font: "Arial"
    }
  }
})

jest.mock("../../src/utils/getRouteName", () => ({
  getRouteName: () => "HOME"
}))

describe("TextField tests", () => {

  it("should render TextField successfully", () => {
    const store = mockStore()({
      validation: { [AppScreens.HOME]: { test: { error: [] } } },
      kiosk: createKioskSettingsState()
    })
    const tree = renderer.create(
      <Provider store={store}>
        <InputWrapper name="test" />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render TextField and pass provided props to it", () => {
    const tree = renderer.create(
      <TextField autoCorrect={false} keyboardType="numeric" />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render TextField wrapped in View", () => {
    const store = mockStore()({
      validation: { [AppScreens.HOME]: { test: { error: [] } } },
      kiosk: createKioskSettingsState()
    })
    const tree = renderer.create(
      <Provider store={store}>
        <InputWrapper name="test" validate={validators.strlen} />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it("should dispatch validation state change action", () => {
  //   const store = mockStore()({
  //     validation: { [AppScreens.HOME]: { test: { error: [] } } }
  //   })
  //   const handler = jest.fn()
  //   const wrapper = shallow(
  //     <InputWrapper
  //       name="test"
  //       validate={validators.strlen}
  //       onEndEditing={handler}
  //     />,
  //     { context: { store } }
  //   ).dive()
  //   expect(wrapper.find(View).length).toBe(2)
  //   expect(wrapper.find(TextField).length).toBe(1)
  //   wrapper.find(TextField).simulate(
  //     "endEditing",
  //     { nativeEvent: { text: "" } }
  //   )
  //   expect(handler).toBeCalled()
  //   expect(store.getActions()).toEqual([{
  //     type: VALIDATION_STATE_CHANGE,
  //     payload: {
  //       [AppScreens.HOME]: {
  //         test: { error: ["String is empty"], valid: false }
  //       }
  //     }
  //   }])
  // })

  // it("should dispatch validation state change action for mobile number field", () => {
  //   const store = mockStore()({
  //     validation: {
  //       [AppScreens.HOME]: {
  //         test: {
  //           error: ["String is empty"], valid: false
  //         }
  //       }
  //     }
  //   })
  //   const wrapper = shallow(
  //     <InputWrapper
  //       name="test"
  //       validate={[validators.strlen, validators.phoneNumber]}
  //     />,
  //     { context: { store } }
  //   ).dive()
  //   expect(wrapper.find(TextField).length).toBe(1)
  //   expect(wrapper.find(View).length).toBe(3)
  //   wrapper.find(TextField).simulate(
  //     "endEditing",
  //     { nativeEvent: { text: "+447921873635" } }
  //   )
  //   expect(store.getActions()).toEqual([{
  //     type: VALIDATION_STATE_CHANGE,
  //     payload: {
  //       [AppScreens.HOME]: {
  //         test: { error: [], valid: true }
  //       }
  //     }
  //   }])
  // })

  // it("should not add new error to the list if such already exists (one validator)", () => {
  //   const store = mockStore()({
  //     validation: {
  //       [AppScreens.HOME]: {
  //         test: {
  //           error: ["String is empty"], valid: false
  //         }
  //       }
  //     }
  //   })
  //   const wrapper = shallow(
  //     <InputWrapper
  //       name="test"
  //       validate={validators.strlen}
  //     />,
  //     { context: { store } }
  //   ).dive()
  //   expect(wrapper.find(TextField).length).toBe(1)
  //   expect(wrapper.find(View).length).toBe(3)
  //   wrapper.find(TextField).simulate(
  //     "endEditing",
  //     { nativeEvent: { text: "" } }
  //   )
  //   expect(store.getActions()).toEqual([{
  //     type: VALIDATION_STATE_CHANGE,
  //     payload: {
  //       [AppScreens.HOME]: {
  //         test: {
  //           error: ["String is empty"], valid: false
  //         }
  //       }
  //     }
  //   }])
  // })

  // it("should not add new error to the list if such already exists (multiple validators)", () => {
  //   const store = mockStore()({
  //     validation: {
  //       [AppScreens.HOME]: {
  //         test: {
  //           error: ["String is empty"], valid: false
  //         }
  //       }
  //     }
  //   })
  //   const wrapper = shallow(
  //     <InputWrapper
  //       name="test"
  //       validate={[validators.strlen, validators.phoneNumber]}
  //     />,
  //     { context: { store } }
  //   ).dive()
  //   expect(wrapper.find(TextField).length).toBe(1)
  //   expect(wrapper.find(View).length).toBe(3)
  //   wrapper.find(TextField).simulate(
  //     "endEditing",
  //     { nativeEvent: { text: "" } }
  //   )
  //   expect(store.getActions()).toEqual([{
  //     type: VALIDATION_STATE_CHANGE,
  //     payload: {
  //       [AppScreens.HOME]: {
  //         test: {
  //           error: ["String is empty"], valid: false
  //         }
  //       }
  //     }
  //   }])
  // })

  // it("should not validate if no validator function provided", () => {
  //   const store = mockStore()({
  //     validation: {
  //       [AppScreens.HOME]: {
  //         test: {
  //           error: ["String is empty"], valid: false
  //         }
  //       }
  //     }
  //   })
  //   const wrapper = shallow(
  //     <InputWrapper name="test" />,
  //     { context: { store } }
  //   )
  //   expect(wrapper.dive().find(TextField).length).toBe(1)
  //   expect(wrapper.dive().find(View).length).toBe(0)
  //   wrapper.dive().find(TextField).simulate(
  //     "endEditing",
  //     { nativeEvent: { text: "" } }
  //   )
  //   expect(store.getActions()).toEqual([])
  // })

  it("should display validated field with popup", () => {
    const store = mockStore()({
      validation: {
        [AppScreens.HOME]: {
          test: {
            error: ["String is empty"], valid: false
          }
        }
      },
      kiosk: createKioskSettingsState()
    })
    const tree = renderer.create(
      <Provider store={store}>
        <InputWrapper
          name="test"
          popupText="popup"
          validate={validators.strlen}
          withPopup
        />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it("should display not validated field with popup", () => {
  //   const store = mockStore()({
  //     validation: {
  //       [AppScreens.HOME]: {
  //         test: {
  //           error: ["String is empty"], valid: false
  //         }
  //       }
  //     }
  //   })
  //   const wrapper = shallow(
  //     <InputWrapper
  //       name="test"
  //       popupText="popup"
  //       withPopup
  //     />,
  //     { context: { store } }
  //   ).dive()
  //   expect(wrapper.find(TextField).length).toBe(1)
  //   expect(wrapper.find(TouchableOpacity).length).toBe(1)
  //   expect(wrapper.find(Popup).length).toBe(1)
  //   expect(wrapper.find(Popup).prop("show")).toBeFalsy()
  //   wrapper.find(TouchableOpacity).simulate("press")
  //   expect(store.getActions()).toEqual([])
  //   expect(wrapper.find(Popup).prop("show")).toBeTruthy()
  // })

})
