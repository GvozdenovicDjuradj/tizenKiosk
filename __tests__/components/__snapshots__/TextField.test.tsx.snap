// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField tests should display validated field with popup 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#dfdfdf",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "width": "100%",
        },
        Object {
          "borderColor": "red",
          "borderWidth": 1,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      dispatch={[Function]}
      name="test"
      onChange={[Function]}
      onEndEditing={[Function]}
      rejectResponderTermination={true}
      screen="HOME"
      style={
        Array [
          Object {
            "borderWidth": 0,
            "flex": 1,
            "fontSize": 26,
            "paddingLeft": 20,
          },
          Array [
            Object {
              "borderWidth": 0,
              "flex": 1,
              "fontSize": 26,
              "paddingLeft": 20,
            },
            undefined,
            Object {},
            Object {
              "height": 54,
            },
          ],
          Object {
            "fontFamily": "Arial",
          },
        ]
      }
      underlineColorAndroid="transparent"
      validate={[Function]}
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#dfdfdf",
            "fontFamily": "FontAwesome5FreeSolid",
            "fontSize": 26,
          }
        }
      >
        ÔÅö
      </Text>
    </View>
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#222",
        "bottom": "100%",
        "justifyContent": "center",
        "left": 0,
        "opacity": 0,
        "paddingHorizontal": 20,
        "position": "absolute",
        "width": "100%",
        "zIndex": 1000,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      popup
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "red",
          "fontSize": 24,
          "textAlign": "center",
        }
      }
    >
      String is empty
    </Text>
  </View>
</View>
`;

exports[`TextField tests should render TextField and pass provided props to it 1`] = `
<TextInput
  allowFontScaling={true}
  autoCorrect={false}
  keyboardType="numeric"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "borderWidth": 0,
        "flex": 1,
        "fontSize": 26,
        "paddingLeft": 20,
      },
      undefined,
      Object {
        "fontFamily": undefined,
      },
    ]
  }
  underlineColorAndroid="transparent"
/>
`;

exports[`TextField tests should render TextField successfully 1`] = `
<TextInput
  allowFontScaling={true}
  dispatch={[Function]}
  name="test"
  rejectResponderTermination={true}
  screen="HOME"
  style={
    Array [
      Object {
        "borderWidth": 0,
        "flex": 1,
        "fontSize": 26,
        "paddingLeft": 20,
      },
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#dfdfdf",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "width": "100%",
        },
        undefined,
        Object {
          "paddingRight": 20,
        },
      ],
      Object {
        "fontFamily": "Arial",
      },
    ]
  }
  underlineColorAndroid="transparent"
/>
`;

exports[`TextField tests should render TextField wrapped in View 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#dfdfdf",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "width": "100%",
        },
        Object {},
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      dispatch={[Function]}
      name="test"
      onChange={[Function]}
      onEndEditing={[Function]}
      rejectResponderTermination={true}
      screen="HOME"
      style={
        Array [
          Object {
            "borderWidth": 0,
            "flex": 1,
            "fontSize": 26,
            "paddingLeft": 20,
          },
          Array [
            Object {
              "borderWidth": 0,
              "flex": 1,
              "fontSize": 26,
              "paddingLeft": 20,
            },
            undefined,
            Object {
              "paddingRight": 20,
            },
            Object {
              "height": 54,
            },
          ],
          Object {
            "fontFamily": "Arial",
          },
        ]
      }
      underlineColorAndroid="transparent"
      validate={[Function]}
    />
  </View>
</View>
`;
